src/btree/bt_cursor.c

```diff
@@ -409,6 +414,15 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
 	WT_STAT_CONN_INCR(session, cursor_search);
 	WT_STAT_DATA_INCR(session, cursor_search);
 
+	WT_CURFILE_OP_PUSH;
+
+	/*
+	 * The pinned page goes away if we do a search, make sure there's a
+	 * local copy of any key, then re-save the cursor state.
+	 */
+	WT_ERR(__cursor_copy_int_key(cursor));
+	WT_CURFILE_OP_PUSH;
```

Comment:
Even with the comment, it seems confusing to "push" twice here.  It's really more "save / restore state" than "push / pop".

Since we're in the area and I had to go back and page this in, I think it would be a lot clearer if `WT_CURFILE_OP_DECL` declared a named variable (an opaque struct) that is passed to inlined `save` and `restore` functions, to make it clear they have some state and operate on the `cursor` local variable.